lightly/api/api_workflow_tags.py

```diff
@@ -73,3 +73,75 @@ def get_filenames_in_tag(
             masked_select_from_list(filenames_on_server)
 
         return filenames_tag
+
+    def create_tag_from_filenames(
+        self,
+        list_of_filenames: List[str],
+        new_tag_name: str,
+        parent_tag_id: str = None
+    ) -> TagData:
+        """Creates a new tag from a list of filenames.
+
+        Args:
+            list_of_filenames:
+                A list of filenames to be included in the new tag.
+            new_tag_name:
+                The name of the new tag.
+            parent_tag_id:
+                The tag defining where to sample from, default: None resolves to the initial-tag.
+
+        Returns:
+            The newly created tag.
+
+        Raises:
+            RuntimeError
+        """
+        
+        # make sure the tag name does not exist yet
+        tags = self.get_all_tags()
+        if new_tag_name in [tag.name for tag in tags]:
+            raise RuntimeError(f'There already exists a tag with tag_name {new_tag_name}.')
+        if len(tags) == 0:
+            raise RuntimeError('There exists no initial-tag for this dataset.')
+
+        # fallback to initial tag if no parent tag is provided
+        if parent_tag_id is None:
+            parent_tag_id = tags[-1].id
+
+        tot_size = tags[-1].tot_size
```

Comment:
I would prefer searching for the tag named 'initial-tag' and take this as the parent tag. Otherwise we might run into subtle bugs we do not find.
It is easy: `parent_tag_id = next(tag.id for tag in tags if tag.name=='initial-tag')`

Reply:
And for `tot_size` I would use` len(frames_server)`, then you don't need to rely on the last tag.